### PR TITLE
`General`: Add default programming language to courses

### PR DESCRIPTION
**Pull request by ge65cer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context

Instructors must select the programming language each time they create a programming exercise.
Generally, a course will use the same programming language from start to finish.
Therefore, the user experience can be improved by choosing a default programming language when creating/editing a course, which will the automatically be pre-set when creating a programming exercise.

### Description

This pull request adds a **new optional attribute** to Courses: `defaultProgrammingLanguage` (String, VARCHAR 255)
On the client, this attribute may be set when creating or updating a Course from the existing list of available programming languages (and an empty default option).
When creating a programming exercise, the programming language will be pre-selected accordingly in the dropdown menu.

> This is one of my first pull requests on this project, so constructive feedback is very welcome.

### Steps for Testing

Prerequisites:
- 1 Instructor

Steps:
1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new Course or select an existing one
4. Choose a default programming language
5. Create a new programming exercise: the corresponding programming language should be pre-selected
7. Make sure all other exercise settings (e.g., template) that depend on the programming language are shown correct

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-update.component.ts | 72% | 89% |
| programming-exercise-update.component.ts | 74% | 77% |

### CODE REVIEW

File: src/main/webapp/i18n/de/course.json
Comment (julian-christl) [inline src/main/webapp/i18n/de/course.json:69]:
```suggestion
            "defaultProgrammingLanguage": "Standard Programmiersprache",
```

File: src/main/webapp/app/course/manage/course-update.component.html
Comment (julian-christl) [inline src/main/webapp/app/course/manage/course-update.component.html:148]: I don't have the option to unset the default value. Please add the option.

**Review by julian-christl — CHANGES_REQUESTED** [review]
Works in general as described, I have one request and two remarks to the code.

**Review by mallgaier — APPROVED** [review]
Tested on TS1.
Code changes look good, functionality as expected. 
Good job 👍

**Review by julian-christl — DISMISSED** [review]
Thanks for the changes. Code approved, test successful.

**Comment by krusche** [comment]
Can you also select an empty programming language in the course?
And please add / adapt some server and client tests so that the new functionality is tested!

**Comment by ge65cer** [comment]
@krusche Yes, an empty (default) option was added as requested by the reviewers. Screenshot updated.

File: src/main/resources/config/liquibase/master.xml
Comment (Dominik-Weinzierl) [inline src/main/resources/config/liquibase/master.xml:159]: To maintain chronology, you should add your change at the end :)

**Review by Dominik-Weinzierl — COMMENTED** [review]
Changes looks good to me 👍  Only one small comment.

File: src/main/resources/config/liquibase/master.xml
Comment (ge65cer) [reply src/main/resources/config/liquibase/master.xml:159]: That must have come from merging `develop`… should I update the timestamp as well?

**Comment by ge65cer** [comment]
Replaced the Cypress test with a unit test for now so that it doesn't block this PR.

**Review by mallgaier — APPROVED** [review]
Re-approve after test replacement and merging into develop

**Pull request merged**
